app/models/transaction.rb

```diff
@@ -1,4 +1,9 @@
 class Transaction < ApplicationRecord
   enum :result, {"success" => 0, "failed" => 1}
+
   belongs_to :invoice
+
+  validates :credit_card_number, presence: true
+  validates :credit_card_expiration_date, presence: true
+  validates :result, presence: true
```

Comment:
Same as the above validations, where I think we want to validate the result as being either "success" or "failed" and not letting it be anything else